include/irr/builtin/material_compiler/glsl/rasterization/impl.glsl

```diff
@@ -0,0 +1,231 @@
+				bxdf_eval_scalar_part = irr_glsl_beckmann_height_correlated_cos_eval_DG(s, uf.isotropic, currInteraction.isotropic, a2);
+#else
+				bxdf_eval_scalar_part = irr_glsl_beckmann_aniso_height_correlated_cos_eval_DG(s, uf, currInteraction, a, a2, ay, ay2);
+#endif
+			}
+
+		} CASE_END
+#endif
+
+#ifdef NDF_PHONG
+		CASE_BEGIN(ndf, NDF_PHONG) {
+
+#if defined(OP_THINDIELECTRIC) || defined(OP_DIELECTRIC)
+			if (is_bsdf) {
+#ifdef ALL_ISOTROPIC_BXDFS
+				bxdf_eval_scalar_part = irr_glsl_beckmann_smith_height_correlated_dielectric_cos_eval_wo_cache_validation(s, currInteraction, uf, ior[0].x, a2);
+#else
+				bxdf_eval_scalar_part = irr_glsl_beckmann_aniso_smith_height_correlated_cos_eval_wo_cache_validation(s, currInteraction, uf, ior[0].x, a, a2, ay, ay2);
+#endif
+			}
+			else
+#endif
+			{
+				float n = irr_glsl_alpha2_to_phong_exp(a2);
+#ifdef ALL_ISOTROPIC_BXDFS
+				bxdf_eval_scalar_part = irr_glsl_blinn_phong_cos_eval_DG(s, uf.isotropic, currInteraction.isotropic, n, a2);
+#else
+				float ny = irr_glsl_alpha2_to_phong_exp(ay2);
+				bxdf_eval_scalar_part = irr_glsl_blinn_phong_cos_eval_DG(s, uf, currInteraction, n, ny, a2, ay2);
+#endif
+			}
+
+		} CASE_END
+#endif
+
+		CASE_OTHERWISE
+		{} //else "empty braces"
+		END_CASES
+	}
+
+	mat2x4 srcs = instr_fetchSrcRegs(instr, regs);
+	bxdf_eval_t result;
+
+	BEGIN_CASES(op)
+#ifdef OP_DIFFUSE
+	CASE_BEGIN(op, OP_DIFFUSE) {
+		result = instr_execute_cos_eval_DIFFUSE(instr, s, params, bsdf_data);
+	} CASE_END
+#endif
+#ifdef OP_CONDUCTOR
+	CASE_BEGIN(op, OP_CONDUCTOR) {
+		result = instr_execute_cos_eval_CONDUCTOR(instr, s, uf, bxdf_eval_scalar_part, params, bsdf_data);
+	} CASE_END
+#endif
+#ifdef OP_PLASTIC
+	CASE_BEGIN(op, OP_PLASTIC) {
+		vec2 dummy;
+		result = instr_execute_cos_eval_PLASTIC(instr, s, uf, bxdf_eval_scalar_part, params, bsdf_data, dummy);
+	} CASE_END
+#endif
+#ifdef OP_COATING
+	CASE_BEGIN(op, OP_COATING) {
+		result = instr_execute_cos_eval_COATING(instr, srcs, params, bsdf_data);
+	} CASE_END
+#endif
+#ifdef OP_DIFFTRANS
+	CASE_BEGIN(op, OP_DIFFTRANS) {
+		result = instr_execute_cos_eval_DIFFTRANS(instr, s, params, bsdf_data);
+	} CASE_END
+#endif
+#ifdef OP_DIELECTRIC
+	CASE_BEGIN(op, OP_DIELECTRIC) {
+		result = instr_execute_cos_eval_DIELECTRIC(instr, s, bxdf_eval_scalar_part);
+	} CASE_END
+#endif
+#ifdef OP_THINDIELECTRIC
+	CASE_BEGIN(op, OP_THINDIELECTRIC) {
+		result = instr_execute_cos_eval_THINDIELECTRIC(instr, s, params, bsdf_data);
+	} CASE_END
+#endif
+#ifdef OP_BLEND
+	CASE_BEGIN(op, OP_BLEND) {
+		result = instr_execute_cos_eval_BLEND(instr, srcs, params, bsdf_data);
+	} CASE_END
+#endif
+#ifdef OP_BUMPMAP
+	CASE_BEGIN(op, OP_BUMPMAP) {
+		instr_execute_BUMPMAP(instr, srcs);
+	} CASE_END
+#endif
+#ifdef OP_SET_GEOM_NORMAL
+	CASE_BEGIN(op, OP_SET_GEOM_NORMAL) {
+		instr_execute_SET_GEOM_NORMAL();
+	} CASE_END
+#endif
+	CASE_OTHERWISE
+	{} //else "empty braces"
+	END_CASES
+
+	if (op_isBXDForBlend(op))
+		writeReg(REG_DST(regs), result);
```

Comment:
this seems like duplicated effort:

https://github.com/buildaworldnet/IrrlichtBAW/pull/424#discussion_r509630949

https://github.com/buildaworldnet/IrrlichtBAW/pull/424#discussion_r509667351